broadinstitute/single_cell_portal_core
# Count result selections per search, for refined CTR metrics (SCP-4291, SCP-4303)

This tracks the number of times search results were clicked _per search_.  It refines UX analytics for global study search.

Previously, if a user searched, say, "human" and clicked 6 results for a given search, e.g. `command` key + click, we could get peculiar average click-through rate (CTR) metrics like "600%".  Here's an example: https://mixpanel.com/s/UpXjw.  Original CTR analytics were instrumented in #1446.

Now, we'll be able to restrict Mixpanel queries by e.g. `numSelections = 1`, giving us more intuitive CTR metrics.  There's a fair chance what we call "CTR" will actually be "first CTR" -- i.e. the rate at which searches were proceeded by selection of _any_ search result.  Having `numSelections` be an integer count rather than a Boolean (e.g. `firstSelection = true`) lets us analyze multi-result-selection UX more deeply in the future, if we want.

Per feedback, this update also consolidates search-related logging.

To test:
1. Open Network panel in DevTools, filter requests to "bard method:POST", go to SCP home page
2. Click a result in the list such that it opens in a new tab, e.g. via `command` key + click on an SCP study title
3. Confirm an event `select-search-result` contains `numSearches: 0` and `numSearchSelections: 1`
4. Do step 2 again
5. Confirm an event `select-search-result` contains `numSearches: 0` and `numSearchSelections: 2`
6. Search "human"
7. Do step 2 again
8. Confirm an event `select-search-result` contains `numSearches: 1` and `numSearchSelections: 1`
9. Confirm an event `select-search-result` contains `numSearches: 1` and `numSearchSelections: 2`

This satisfies SCP-4291 and SCP-4303.
